### ISSUE TITLE
Does not evaluate Terraform null variable default values

### ISSUE THREAD

**Issue by jdheyburn** [bridgecrewio/checkov#651]
**Describe the bug**
I noticed this for CKV_AWS_86 "Ensure Cloudfront distribution has Access Logging enabled", unsure if it exists elsewhere for other checks. 

Checkov does not seem to evaluate Terraform `null` default values for variables in modules. Null in Terraform is the equivalent of having the attribute being set to that value as absent. 

I believe this is causing false negatives to appear in reports (passing when they should fail).

**To Reproduce**

main.tf
```hcl
resource "aws_cloudfront_distribution" "cf_dis" {
  # ...

  logging_config {
    include_cookies = var.logging_include_cookies
    bucket          = var.logging_bucket_id
    prefix          = var.logging_bucket_prefix
  }
 # ...
}
```

variables.tf
```hcl
variable "logging_include_cookies" {
  type        = bool
  description = "Whether to enable cookies in access logging"
  default     = null
}

variable "logging_bucket_id" {
  type        = string
  description = "The bucket ID where to store access logs"
  default     = null
}

variable "logging_bucket_prefix" {
  type        = string
  description = "The prefix where to store access logs"
  default     = null
}
```

The current settings means that no logging config is actually set, when when running:
```
$ checkov -d . -c CKV_AWS_86
```

It passes:

```
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
By bridgecrew.io | version: 1.0.612 

terraform scan results:

Passed checks: 1, Failed checks: 0, Skipped checks: 0

Check: CKV_AWS_86: "Ensure Cloudfront distribution has Access Logging enabled"
        PASSED for resource: aws_cloudfront_distribution.cf_dis
```

**Expected behavior**
I would expect Checkov to evaluate the `null` value and fail the check.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS
 - Checkov Version [e.g. 22]: 1.0.612

**Additional context**
Additionally, even if the default value was set to a value, the calling module could overwrite the attribute with `null` - causing another false negative. I am not sure if it is related to this issue I am describing though.

On the other hand, the null check seems to work fine for S3 versioning for the below, where setting it to null causes a failed check.

main.tf
```hcl
resource "aws_s3_bucket" "website_bucket" {
  versioning {
      enabled = var.versioning
  }
}
```

variables.tf
```hcl
variable "versioning" {
    default = null
}
```

**Comment by jdheyburn** [comment]
So I made a local build changing `CloudfrontDistributionLogging` to be a `BaseResourceCheck` instead of `BaseResourceValueCheck`:

```python
class CloudfrontDistributionLogging(BaseResourceCheck):

    def __init__(self):
        name = "Ensure Cloudfront distribution has Access Logging enabled"
        id = "CKV_AWS_86"
        supported_resources = ['aws_cloudfront_distribution']
        categories = [CheckCategories.LOGGING]
        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)

    def scan_resource_conf(self, conf):
        parent_key = 'logging_config'
        if parent_key not in conf.keys():
            return CheckResult.FAILED
        key = 'bucket'
        if key not in conf[parent_key][0]:
            return CheckResult.FAILED
        if not conf[parent_key][0][key][0]:
            return CheckResult.FAILED
        return CheckResult.PASSED
```

> `null` Terraform values appear as `None` in checkov.

When I set `bucket = null` this is correctly failed by checkov, however when the attribute is set to a variable such as `var.logging_bucket_id` then this does not appear as `None`, even though the default value for the variable is `null`.

I suppose I am a bit confused as given the [Evaluation documentation](https://github.com/bridgecrewio/checkov/blob/master/docs/2.Concepts/Evaluations.md) I would expect it to be evaluated as `null`. 

Is there something I'm missing?

**Comment by jdheyburn** [comment]
I think the issue is with this line, or previous:
https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/evaluation/evaluation_methods/const_variable_evaluation.py#L89

The default `null` is not parsed to be a default `None`.

**Comment by angatha** [comment]
@jdheyburn for myself I guessed that `null` is used to indicate that the default value should be used but I didn't confirm it. Do you know some source that clearly states that or if it is just a convention?

**Comment by jdheyburn** [comment]
@angatha see the below doc, which indicates `null` is the same as omission:
https://www.terraform.io/docs/configuration/expressions.html#null
